tests/unit/models/test_exercise_analysis.py

```
@@ -0,0 +1,155 @@
+        )
+        
+        assert result.exercise_type == "unknown"
+        assert result.calories_burned == 0
+        assert result.duration == "unknown"
+        assert result.intensity == "unknown"
+        assert result.error == "Failed to analyze exercise"
+    
+    def test_exercise_analysis_result_validation(self):
+        """Test validation of required fields."""
+        # All fields are required
+        with pytest.raises(Exception):
```

Ruff (0.8.2)

`pytest.raises(Exception)` should be considered evil

(B017)